src/renderers/shaders/UniformsLib.js

```diff
@@ -6,16 +7,162 @@ import { Matrix3 } from '../../math/Matrix3.js';
  * Uniforms library for shared webgl shaders
  */
 
+function refreshUVTransform2D( uniforms, material ) {
+
+	// uv repeat and offset setting priorities
+	// 1. color map
+	// 2. alpha map
+
+	var uvScaleMap;
+
+	if ( material.map ) {
+
+		uvScaleMap = material.map;
+
+	} else if ( material.alphaMap ) {
+
+		uvScaleMap = material.alphaMap;
+
+	}
+
+	if ( uvScaleMap !== undefined ) {
+
+		if ( uvScaleMap.matrixAutoUpdate === true ) {
+
+			uvScaleMap.updateMatrix();
+
+		}
+
+		uniforms.uvTransform.value.copy( uvScaleMap.matrix );
+
+	}
+
+}
+
 var UniformsLib = {
 
 	common: {
 
-		diffuse: { value: new Color( 0xeeeeee ) },
+		diffuse: { value: new Color( 0xeeeeee ), onUpdate: function( uniforms, material ) {
+
+			if ( material.color ) {
+
+				uniforms.diffuse.value.copy( material.color );
+				
+			}
+
+		} },
```

Comment:
Nit: suggested formatting pattern:
```js
var UniformsLib = {

	common: {

		diffuse: {

			value: new Color( 0xeeeeee ),

			onUpdate: function ( uniforms, material ) {

				if ( material.color ) {

					uniforms.diffuse.value.copy( material.color );

				}

			}

		},

		opacity: { value: 1.0 },

		map: { value: null },

		uvTransform: {

			value: new Matrix3(),

			onUpdate: function ( uniforms, material ) {

				// uv repeat and offset setting priorities
				// 1. color map
				// 2. specular map
				// 3. normal map
				// 4. bump map
				// 5. alpha map
				// 6. emissive map

				var uvScaleMap;

				if ( material.map ) {

					uvScaleMap = material.map;

				} else if ( material.specularMap ) {

					uvScaleMap = material.specularMap;

				} else if ( material.displacementMap ) {

					uvScaleMap = material.displacementMap;

				} else if ( material.normalMap ) {

					uvScaleMap = material.normalMap;

				} else if ( material.bumpMap ) {

					uvScaleMap = material.bumpMap;

				} else if ( material.roughnessMap ) {

					uvScaleMap = material.roughnessMap;

				} else if ( material.metalnessMap ) {

					uvScaleMap = material.metalnessMap;

				} else if ( material.alphaMap ) {

					uvScaleMap = material.alphaMap;

				} else if ( material.emissiveMap ) {

					uvScaleMap = material.emissiveMap;

				}

				if ( uvScaleMap !== undefined ) {

					// backwards compatibility

					if ( uvScaleMap.isWebGLRenderTarget ) {

						uvScaleMap = uvScaleMap.texture;

					}

					if ( uvScaleMap.matrixAutoUpdate === true ) {

						uvScaleMap.updateMatrix();

					}

					uniforms.uvTransform.value.copy( uvScaleMap.matrix );

				}

			}
		},
```

Reply:
This can help you pass the formatting test: https://zz85.github.io/mrdoobapproves/